File: backend/pkg/runtime/status.py
from __future__ import annotations

from enum import StrEnum


class WorkerStatus(StrEnum):
    REGISTERING = "REGISTERING"
    ONLINE = "ONLINE"
    BUSY = "BUSY"
    DRAINING = "DRAINING"
    OFFLINE = "OFFLINE"
    ERROR = "ERROR"


class InferenceStatus(StrEnum):
    PENDING = "pending"
    SCANNING = "scanning"
    BLOCKED = "blocked"
    REVIEW = "review"
    ROUTING = "routing"
    COMPLETED = "completed"
    FAILED = "failed"


class ModelStatus(StrEnum):
    LOADING = "LOADING"
    READY = "READY"
    UNLOADING = "UNLOADING"
    ERROR = "ERROR"

File: backend/pkg/runtime/status.go
package runtime

const (
	WorkerStatusRegistering = "REGISTERING"
	WorkerStatusOnline      = "ONLINE"
	WorkerStatusBusy        = "BUSY"
	WorkerStatusDraining    = "DRAINING"
	WorkerStatusOffline     = "OFFLINE"
	WorkerStatusError       = "ERROR"

	InferenceStatusPending   = "pending"
	InferenceStatusScanning  = "scanning"
	InferenceStatusBlocked   = "blocked"
	InferenceStatusReview    = "review"
	InferenceStatusRouting   = "routing"
	InferenceStatusCompleted = "completed"
	InferenceStatusFailed    = "failed"

	ModelStatusLoading   = "LOADING"
	ModelStatusReady     = "READY"
	ModelStatusUnloading = "UNLOADING"
	ModelStatusError     = "ERROR"
)
